app/ore/project/io/PluginFileData.scala

```diff
@@ -0,0 +1,216 @@
+  extends DataValue[List[Dependency]](key, value)
+
+sealed abstract case class FileType(fileName: String) {
+  def getData(bufferedReader: BufferedReader): List[DataValue[_]]
+}
+
+object ModInfo extends FileType("mcmod.info") {
+  override def getData(bufferedReader: BufferedReader): List[DataValue[_]] = {
+
+    val dataValues = new ListBuffer[DataValue[_]]
+
+    try {
+      val parser = new JsonParser().parse(new JsonReader(bufferedReader))
+
+      val modArray = parser.getAsJsonArray
+      for (i <- 0 until modArray.size()) {
+        val modObject = modArray.get(i).getAsJsonObject
+
+        // collect members
+
+        if (modObject.has("modid"))
+          dataValues += StringDataValue("id", modObject.get("modid").getAsString)
+        if (modObject.has("name"))
+          dataValues += StringDataValue("name", modObject.get("name").getAsString)
+        if (modObject.has("version"))
+          dataValues += StringDataValue("version", modObject.get("version").getAsString)
+        if (modObject.has("description"))
+          dataValues += StringDataValue("description", modObject.get("description").getAsString)
+        if (modObject.has("url"))
+          dataValues += StringDataValue("url", modObject.get("url").getAsString)
+
+        if (modObject.has("authors")) {
+          val authors = new ListBuffer[String]
+          val authorsJson = modObject.getAsJsonArray("authors")
+          for (j <- 0 until authorsJson.size()) {
+            val element = authorsJson.get(j)
+            authors += element.getAsString
+          }
+          dataValues += StringListValue("authors", authors.toList)
+        }
+
+        // collect dependencies
+
+        val dependencies = new ListBuffer[Dependency]
+
+        if (modObject.has("requiredMods")) {
+          val dependenciesJson = modObject.getAsJsonArray("requiredMods")
+          for (j <- 0 until dependenciesJson.size()) {
+            var dependencyId = ""
+            var dependencyVersion = ""
+
+            val dependencyJson = dependenciesJson.get(j)
+            val parts = dependencyJson.getAsString.split("@")
+            if (parts.nonEmpty) {
+              dependencyId = parts(0)
+            }
+            if (parts.length > 1) {
+              dependencyVersion = parts(1)
+            }
+
+            if (dependencyId != null && !dependencyId.isEmpty) {
+              dependencies += Dependency(dependencyId, dependencyVersion)
+            }
+          }
+        }
+
+        dataValues += DependencyDataValue("dependencies", dependencies.toList)
+      }
+    } catch {
+      case e: Exception => e.printStackTrace()
+    }
+
+    dataValues.toList
+  }
+}
+
+object Manifest extends FileType("META-INF/MANIFEST.MF") {
+  override def getData(bufferedReader: BufferedReader): List[DataValue[_]] = {
+    val dataValues = new ListBuffer[DataValue[_]]
+
+    val lines = Stream.continually(bufferedReader.readLine()).takeWhile(_ != null)
```

Comment:
Stream is considered one of the "bad" things in the scala library. Use NIO instead.